modin/pandas/dataframe.py

```diff
@@ -1090,7 +1090,20 @@ def cumsum(self, axis=None, skipna=True, *args, **kwargs):
             The cumulative sum of the DataFrame.
         """
         axis = pandas.DataFrame()._get_axis_number(axis) if axis is not None else 0
-        self._validate_dtypes(numeric_only=True)
+        dtype = self.dtypes[0]
+        # Checks to make sure that all dtypes are of the same category as the first
+        if axis and not all(
+            (is_numeric_dtype(dtype) and is_numeric_dtype(t))
+            or (
+                dtype == np.dtype("timedelta64[ns]")
+                and t == np.dtype("timedelta64[np]")
```

Comment:
Should it be `ns` instead of `np`?
```suggestion
                and t == np.dtype("timedelta64[ns]")
```